snap/component_test.go

```diff
@@ -307,3 +308,144 @@ func (s *componentSuite) TestComponentSideInfoEqual(c *C) {
 		c.Check(csi.Equal(tc.csi), Equals, tc.equal)
 	}
 }
+
+func (s *componentSuite) TestReadComponentInfoFinishedWithSnapInfo(c *C) {
+	const componentYaml = `component: snap+component
+type: test
+version: 1.0
+summary: short description
+description: long description
+`
+	const compName = "snap+component"
+	testComp := snaptest.MakeTestComponentWithFiles(c, compName+".comp", componentYaml, [][]string{
+		{"meta/hooks/install", "echo 'explicit hook'"},
+		{"meta/hooks/pre-refresh", "echo 'implicit hook'"},
+	})
+
+	compf, err := snapfile.Open(testComp)
+	c.Assert(err, IsNil)
+
+	const snapYaml = `
+name: snap
+components:
+  component:
+    type: test
+    hooks:
```

Comment:
nitpick: maybe this test should contain more than one hook, with and without plugs, etc.